packages/reports/src/Containers/__tests__/reports.spec.tsx

```diff
@@ -0,0 +1,193 @@
+                </Router>
+            </StoreProvider>
+        );
+    };
+    test('renders Reports component', () => {
+        const history = createMemoryHistory();
+        renderReports(store, history);
+        expect(screen.getAllByText('Report 1').length).toBeGreaterThan(0);
+        expect(screen.getAllByText('Report 2').length).toBeGreaterThan(0);
+    });
+
+    test('shows loader when is_logged_in is false', () => {
+        const history = createMemoryHistory();
+        store = mockStore({
+            ...mock,
+            client: {
+                ...mock.client,
+                is_logged_in: false,
+                is_logging_in: true,
+            },
+        });
+        renderReports(store, history);
+        expect(screen.getByText('Loading...')).toBeInTheDocument();
+    });
+
+    test('tracks Analytics events on open and close', () => {
+        const history = createMemoryHistory();
+        const { unmount } = render(
+            <StoreProvider store={store}>
+                <Router history={history}>
+                    <Reports history={history} location={history.location} routes={routes} />
+                </Router>
+            </StoreProvider>
+        );
+
+        expect(Analytics.trackEvent).toHaveBeenCalledWith(
+            'ce_reports_form',
+            expect.objectContaining({ action: 'open' })
+        );
+        unmount();
+        expect(Analytics.trackEvent).toHaveBeenCalledWith(
+            'ce_reports_form',
+            expect.objectContaining({ action: 'close' })
+        );
+    });
+
+    test('navigates to a different route on select change', () => {
+        const history = createMemoryHistory();
+        renderReports(store, history);
+
+        fireEvent.change(screen.getByRole('combobox'), { target: { value: '/report2' } });
+        expect(history.location.pathname).toBe('/report2');
+    });
+
+    test('calls routeBackInApp on close button click', () => {
+        const mockRouteBackInApp = jest.fn();
+        store = mockStore({
+            ...mock,
+            common: {
+                ...mock.common,
+                routeBackInApp: mockRouteBackInApp,
+            },
+        });
+        const history = createMemoryHistory();
+        renderReports(store, history);
+        fireEvent.click(screen.getByTestId('onclose-click'));
+        expect(mockRouteBackInApp).toHaveBeenCalled();
+    });
+
+    test('sets vertical_tab_index to 0 if the selected route is default', () => {
+        const history = createMemoryHistory();
+        history.push('/report1');
+        renderReports(store, history);
+
+        expect(mockVerticalTab).toHaveBeenCalledWith(
+            expect.objectContaining({
+                vertical_tab_index: 0,
+            })
+        );
+    });
+
+    test('sets vertical_tab_index to reports_route_tab_index if the selected route is not default', () => {
+        const history = createMemoryHistory();
+        history.push('/report2');
```

Comment:
just a recommendation for maintainability: Would it be possible to put repeatedly used strings like '/report1', '/report2', 'onclose-click', 'Report 1', 'Report 2' into variables? :)

Reply:
done , thanks